shm/src/ship.cpp

```diff
@@ -16,11 +19,52 @@ void Ship::setName(const std::string& name) {
     name_ = name;
 }
 
-Ship& Ship::operator-=(size_t num) {
-    crew_ -= num;
+Ship& Ship::operator-=(size_t crew) {
+    crew_ -= crew;
     return *this;
 }
-Ship& Ship::operator+=(size_t num) {
-    crew_ += num;
+Ship& Ship::operator+=(size_t crew) {
+    crew_ += crew;
     return *this;
 }
+
+void Ship::load(std::shared_ptr<Cargo> cargo) {
+    size_t allAmount = 0;
+    for (auto el : cargo_) {
+        allAmount + el->getAmount();
```

Comment:
Shouldn't it be allAmount += el->getAmount();? '=' missing according to me. You could also add const & to auto el

Reply:
Of course I missed =. I think shared_ptr<Cargo> can be copied without losing time. But I can add const & for safety.